Add note in RPC docs about retries.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73601

Some users had questions about how the RPC framework deals with
failures and whether we retry. Adding a note about this to our docs to
elaborate on our current behavior and why we chose that approach.

![Screen Shot 2022-03-01 at 12 35 56 PM](https://user-images.githubusercontent.com/9958665/156244988-fecd80e2-71ab-4fac-9137-0d3c07578a50.png)

Differential Revision: [D34560199](https://our.internmc.facebook.com/intern/diff/D34560199/)

pritamdamania committed:
Add note in RPC docs about retries.

Some users had questions about how the RPC framework deals with
failures and whether we retry. Adding a note about this to our docs to
elaborate on our current behavior and why we chose that approach.

Differential Revision: [D34560199](https://our.internmc.facebook.com/intern/diff/D34560199/)

[ghstack-poisoned]

satgera commented on docs/source/rpc.rst:
```diff
@@ -190,6 +190,14 @@ Example::
+  :meth:`~torch.distributed.rpc.remote` calls. The reason being that there is 
+  no way the RPC framework can determine whether an operation is idempotent or 
+  not and whether it is safe to retry. As a result, it is the application's 
+  responsibility to deal with failures and retry if necessary.
```
It may help to add some details in what scenario a failure may happen ? eg: RPC communication is based on TCP connections and for a subset of network failures, users may need to retry a call with reasonable backoffs.

mrshenli reviewed (approved):
+1 to @satgera's comment

Stamp to unblock.

pritamdamania committed:
Update on "Add note in RPC docs about retries."


Some users had questions about how the RPC framework deals with
failures and whether we retry. Adding a note about this to our docs to
elaborate on our current behavior and why we chose that approach.

![Screen Shot 2022-03-01 at 12 35 56 PM](https://user-images.githubusercontent.com/9958665/156244988-fecd80e2-71ab-4fac-9137-0d3c07578a50.png)

Differential Revision: [D34560199](https://our.internmc.facebook.com/intern/diff/D34560199/)

[ghstack-poisoned]
